packages/core/src/account/schema.ts

```diff
@@ -56,3 +56,21 @@ export const BaseSmartAccountParamsSchema = <
     }),
     accountAddress: zAddress.optional(),
   });
+
+export const SimpleSmartAccountParamsSchema = <
+  TTransport extends SupportedTransports = Transport
+>() =>
+  BaseSmartAccountParamsSchema<TTransport>().extend({
+    owner: z
+      .any()
+      .refine<SmartAccountSigner>((signer): signer is SmartAccountSigner => {
+        return (
+          typeof signer === "object" &&
+          "signerType" in signer &&
+          "signMessage" in signer &&
+          "signTypedData" in signer &&
+          "getAddress" in signer
```

Comment:
ditto: can we use https://zod.dev/?id=instanceof instead?

Reply:
yeah good q. I tried this but this only works on classes, not types/interfaces, which is what I'm doing here.

here's a min repro showing `instanceOf` didn't work:
<img width="1003" alt="image" src="https://github.com/alchemyplatform/aa-sdk/assets/43521356/ae7ec7d2-0a39-4d1d-988c-8623fe0e6d7b">

Reply:
commented in a previous PR but let's define this is a shareable schema in the code because we do this a lot